src/unix/core.c

```diff
@@ -983,3 +984,61 @@ int uv__dup2_cloexec(int oldfd, int newfd) {
     return r;
   }
 }
+
+
+int uv_os_homedir(char* buffer, size_t* size) {
+  struct passwd pw;
+  struct passwd* result;
+  char* buf;
+  uid_t uid;
+  size_t bufsize;
+  size_t len;
+  int r;
+
+  if (buffer == NULL || size == NULL || *size == 0)
+    return -EINVAL;
+
+  bufsize = sysconf(_SC_GETPW_R_SIZE_MAX);
+
+  if (bufsize <= 0)
+    return -EIO;
+
+  uid = getuid();
```

Comment:
> I also commented on that. :-)

Ha!



> How does python deal with that?

It doesn't. I guess the GIL helps. os.environ is a dictionary which gets loaded when Python and gets updated along the way.

FWIW I *think* I could live with getenv.